homeassistant/components/zha/core/channels/__init__.py

```diff
@@ -29,20 +31,25 @@
     device as zha_core_device,
     discovery as zha_disc,
     registries as zha_regs,
-    typing as zha_typing,
 )
 
-ChannelsDict = dict[str, zha_typing.ChannelType]
+if TYPE_CHECKING:
+    from ...entity import ZhaEntity
+    from ..device import ZHADevice
```

Comment:
Just wondering, is this a best practice for Python / HA codestyle? I haven't seen this yet before and most of the time I just import everything directly, even the imports that are only required for my types.

Reply:
Usually importing it directly is fine. The issue with zha are circular imports which the `if TYPE_CHECKING` guard helps to avoid. Especially as I'm adding typing in steps, an import which might work at first, can still break later.

Reply:
Thanks! Learned something new again 😀